### DISCUSSION TITLE
Could I use `source` value in SystemAccessControl, like prestodb?

### DISCUSSION THREAD

**Discussion by woowahan-jaehoon** [trinodb/trino#9119] [Q&A]
Hi there.

I need `source` value in SystemAccessControl because I need to control to allow some schemas each platform that use trino.

I found this feature is already applied in prestodb.  
https://github.com/prestodb/presto/commit/efd979afa5efcf616ac1ae0b806fcad3a0cf1ab0

Do you have a plan to apply this feature?  
or Could you introduce alternative method?

**Comment by electrum** [comment]
Using `source` is problematic since that is part of the query context, but wouldn't be available within a view (which by default run as the owner of the view and not the user running the query). `SystemSecurityContext` contains `Identity` which has an optional `Principal`, which you could use to differentiate between the system accessing Trino by having them provide different principals.

**Reply by hashhar** [reply]
Also source can easily be spoofed since it's an HTTP header and there are no "access controls" for it. It's main purpose (AFAIK) is to allow informational tagging.

**Reply by woowahan-jaehoon** [reply]
I need to check user's username/group and also check where users run queries so I think that using `source` is one of ways.

I need to check also where users run query because a reason of this situation.
GroupA is allowed to access some tables and GroupB is allowed to access all tables.
User1 is member of GroupA and User2 is member of GroupB.
PlatformA is allowed to access GroupA/GroupB and PlatformB is allowed to access only GroupB.

Then, If PlatformA has result of queries, User1 can read query result of tables that isn't allowed to GroupA after User2 used PlatformA because User2 can use PlatformA to access tables that isn't allowed to GroupA. So I need to block tables that isn't allowed GroupA in PlatformA.
Some platform has access control of query result but user can mistake to set access control.

I found extraCredential in `Identity` class so I think that I can use it instead of `source`.

**Reply by woowahan-jaehoon** [reply]
> Also source can easily be spoofed since it's an HTTP header and there are no "access controls" for it. It's main purpose (AFAIK) is to allow informational tagging.

'Easily be spoofed' is not problem in my case because only some platform can access trino and connection information is preseted in these platforms like zeppelin, redash.
